src/mark.c

```diff
@@ -141,6 +143,17 @@ gboolean r_mark_active(RaucSlot *slot, GError **error)
 	r_slot_status_load(slot);
 	slot_state = slot->status;
 
+	/* It is more sensible to unlock the counter, before marking it primary. Reversing the order would create
+	 * an unintended corner case where a power cut could leave the system configured to boot from a new,
+	 * untested slot without the ability to fall back if that slot fails to boot properly, as it would
+	 * still be locked. So we prepary the safety mechanisms first and then do the potentially risky operation. */
```

Comment:
```suggestion
	 * still be locked. So we prepare the safety mechanisms first and then do the potentially risky operation. */
```

'safe' and 'risky' sounds a bit inappropriate for the individual operations.

Reply:
you're right, and also the information probably is also a bit redundant when taking the previous sentence into account.
```suggestion
	 * still be locked. So it is better to prepare the fallback mechanisms first. */
```